packages/mobile/src/screens/change-email-screen/ChangeEmailScreen.tsx

```diff
@@ -0,0 +1,184 @@
+import { VerifyEmailSubScreen } from '../change-password-screen/SubScreens'
+
+import { ConfirmPasswordSubScreen, NewEmailSubScreen } from './SubScreens'
+
+const messages = {
+  change: 'Change Email',
+  continue: 'Continue',
+  success: 'Email updated!'
+}
+
+const useStyles = makeStyles(({ palette, spacing }) => ({
+  screen: {
+    justifyContent: 'space-between'
+  },
+  bottomSection: {
+    overflow: 'hidden',
+    height: 'auto',
+    padding: spacing(4),
+    paddingBottom: spacing(12),
+    backgroundColor: palette.white,
+    borderTopWidth: 1,
+    borderTopColor: palette.neutralLight6
+  }
+}))
+
+const Stack = createNativeStackNavigator()
+
+const ChangeEmailHeaderLeft = ({ page }: { page: ChangeEmailPage }) => {
+  const navigation = useNavigation()
+  if (page === ChangeEmailPage.VerifyEmail) {
+    return <BackButton />
+  } else {
+    return (
+      <TouchableOpacity
+        onPress={() => navigation.navigate('AccountSettingsScreen')}
+      >
+        <IconClose size='l' color='subdued' />
+      </TouchableOpacity>
+    )
+  }
+}
+
+const ChangeEmailNavigator = ({
+  page,
+  setPage
+}: {
+  page: ChangeEmailPage
+  setPage: (page: ChangeEmailPage) => void
+}) => {
+  const { handleSubmit, isSubmitting } = useFormikContext()
+  const styles = useStyles()
+  const navigation = useNavigation()
+
+  // Only show the back button on the OTP page
+  const screenOptions = useAppScreenOptions({
+    headerRight: () => null,
+    headerLeft: () => {
+      return <ChangeEmailHeaderLeft page={page} />
+    }
+  })
+
+  // Map hook page state to screen navigations
+  useEffect(() => {
+    navigation.navigate(ChangeEmailPage[page])
+  }, [page, navigation])
+
+  // Map navigations back to the hook page state
+  const stateListener = useCallback<
+    EventListenerCallback<EventMapCore<NavigationState>, 'state'>
+  >(
+    (e) => {
+      const state = e.data.state
+      const route = state.routes[state.index]
+      const newPage = ChangeEmailPage[route?.name]
+      if (newPage !== page) {
+        setPage(newPage)
+      }
+    },
+    [page, setPage]
+  )
+
+  return (
+    <Screen variant='secondary' style={styles.screen}>
+      <ScreenContent>
+        <Stack.Navigator
+          screenOptions={screenOptions}
+          screenListeners={{
+            state: stateListener
+          }}
+        >
+          <Stack.Screen
+            name={ChangeEmailPage[ChangeEmailPage.ConfirmPassword]}
+            component={ConfirmPasswordSubScreen}
+          />
+          <Stack.Screen
+            name={ChangeEmailPage[ChangeEmailPage.NewEmail]}
+            component={NewEmailSubScreen}
+          />
+          <Stack.Screen
+            name={ChangeEmailPage[ChangeEmailPage.VerifyEmail]}
+            component={VerifyEmailSubScreen}
+          />
+        </Stack.Navigator>
+        <KeyboardAvoidingView
+          style={styles.bottomSection}
+          keyboardShowingOffset={32}
```

Comment:
unlike some of the other screens, this one keeps the button above the keyboard. That's why only the button is in the avoiding view

Reply:
yay this is brilliant, will copy in some other cases, thank you for figuring this out

Reply:
does this button stay at the bottom when no keyboard present?

Reply:
yes it should!